query_build.go

```diff
@@ -0,0 +1,84 @@
+package bob
+
+import (
+	"fmt"
+	"io"
+)
+
+func BindNamedArgs(buildArgs []any, args []any) ([]any, error) {
+	var nargs []NamedArgument
+	hasNonNamed := false
+	for _, buildArg := range buildArgs {
+		if na, ok := buildArg.(NamedArgument); ok {
+			nargs = append(nargs, na)
+		} else {
+			hasNonNamed = true
+		}
+	}
+	if len(nargs) == 0 {
+		return args, nil
+	}
+	if hasNonNamed {
+		return nil, fmt.Errorf("cannot mix named and non-named arguments")
+	}
```

Comment:
tecnically it wouldn't be a problem mixing named and non-named parameters, I could just forward the parameter in the same order if it wasn't a `NamedArgument`.

Reply:
But how will that work with prepared statements?

Reply:
`mergeNamedArguments` would receive `[]any`, and in [this for loop](https://github.com/stephenafamo/bob/pull/66/files#diff-a1785b1d4282adef1513787b42f45fb946e31f6f6a3c9c4953a50c8e4302df6cR38), I would check if the type is `NamedArgument`, otherwise I would just add the value itself to the return array at the same position.

Reply:
for example, given:

```go
	query := psql.Insert(
		im.Into("actor", "first_name", "last_name"),
                im.Values(psql.Arg(12, psql.NamedArg("in"))),
	)
```

`mergeNamedArguments` would receive as first parameter:

`[]any{12, NamedArg{"in"}}`

in the for loop I would just forward the value that is not NamedArg and only replace the other one, so in:

```go

	queryStr, args, err := bob.BuildWithNamedArgs(query, map[string]any{
		"in": "LAST_NAME",
	})
```

`args` would be `[]any{12, "LAST_NAME"}`.